From b431680ea7a8f8ffbee1995fb951f973a60d9d4e Mon Sep 17 00:00:00 2001
From: Ferdinand Xu <cheng.a.xu@intel.com>
Date: Mon, 13 Jul 2015 21:49:25 -0400
Subject: [PATCH 0686/1164] CDH-42366 : HIVE-11236 BeeLine-Cli: use the same
 output format as old CLI in the new CLI(Ferdinand
 Xu, reviewed by Xuefu Zhang)

Change-Id: Ic1971620ff34a7698405eb46916103fd1eb28221
---
 .../src/java/org/apache/hive/beeline/BeeLine.java  |    3 +++
 .../org/apache/hive/beeline/cli/TestHiveCli.java   |   12 ++++++------
 2 files changed, 9 insertions(+), 6 deletions(-)

diff --git a/beeline/src/java/org/apache/hive/beeline/BeeLine.java b/beeline/src/java/org/apache/hive/beeline/BeeLine.java
index 9eff2b4..29396cd 100644
--- a/beeline/src/java/org/apache/hive/beeline/BeeLine.java
+++ b/beeline/src/java/org/apache/hive/beeline/BeeLine.java
@@ -873,6 +873,9 @@ public int begin(String[] args, InputStream inputStream) throws IOException {
           return code;
         }
         getOpts().updateBeeLineOptsFromConf();
+        getOpts().setShowHeader(false);
+        getOpts().setOutputFormat("dsv");
+        getOpts().setDelimiterForDSV(' ');
       }
 
       if (getOpts().isHelpAsked()) {
diff --git a/beeline/src/test/org/apache/hive/beeline/cli/TestHiveCli.java b/beeline/src/test/org/apache/hive/beeline/cli/TestHiveCli.java
index fc8ed0c..542f1ee 100644
--- a/beeline/src/test/org/apache/hive/beeline/cli/TestHiveCli.java
+++ b/beeline/src/test/org/apache/hive/beeline/cli/TestHiveCli.java
@@ -41,11 +41,11 @@
   private static final int ERRNO_OTHER = 2;
 
   private final static String SOURCE_CONTEXT =
-      "create table if not exists test.testSrcTbl(a string, b string);";
+      "create table if not exists test.testSrcTbl(sc1 string);";
   private final static String SOURCE_CONTEXT2 =
-      "create table if not exists test.testSrcTbl2(a string);";
+      "create table if not exists test.testSrcTbl2(sc2 string);";
   private final static String SOURCE_CONTEXT3 =
-      "create table if not exists test.testSrcTbl3(a string);";
+      "create table if not exists test.testSrcTbl3(sc3 string);";
   final static String CMD =
       "create database if not exists test;\ncreate table if not exists test.testTbl(a string, b "
           + "string);\n";
@@ -113,7 +113,7 @@ public void testDatabaseOptions() {
   @Test
   public void testSourceCmd() {
     File f = generateTmpFile(SOURCE_CONTEXT);
-    verifyCMD("source " + f.getPath() + ";" + "desc testSrcTbl;\nquit;\n", "col_name", os,
+    verifyCMD("source " + f.getPath() + ";" + "desc testSrcTbl;\nquit;\n", "sc1", os,
         new String[] { "--database", "test" }, ERRNO_OK);
     f.delete();
   }
@@ -121,7 +121,7 @@ public void testSourceCmd() {
   @Test
   public void testSourceCmd2() {
     File f = generateTmpFile(SOURCE_CONTEXT3);
-    verifyCMD("source " + f.getPath() + ";" + "desc testSrcTbl3;\nquit;\n", "col_name", os,
+    verifyCMD("source " + f.getPath() + ";" + "desc testSrcTbl3;\nquit;\n", "sc3", os,
         new String[] { "--database", "test" }, ERRNO_OK);
     f.delete();
   }
@@ -159,7 +159,7 @@ public void testVariablesForSource() {
     File f = generateTmpFile(SOURCE_CONTEXT2);
     verifyCMD(
         "set hiveconf:zzz=" + f.getAbsolutePath() + ";\nsource ${hiveconf:zzz};\ndesc testSrcTbl2;",
-        "col_name", os, new String[] { "--database", "test" }, ERRNO_OK);
+        "sc2", os, new String[] { "--database", "test" }, ERRNO_OK);
     f.delete();
   }
 
-- 
1.7.9.5

